drivers/regulator/regulator_da1469x.c

```diff
@@ -0,0 +1,451 @@
+		break;
+	case V14:
+		DCDC_REG_SET(V14, CUR_LIM_MAX_HV, ((max_ua / 30000) - 1));
+		DCDC_REG_SET(V14, CUR_LIM_MAX_LV, ((max_ua / 30000) - 1));
+		DCDC_REG_SET(V14, CUR_LIM_MIN, ((max_ua / 30000) - 1));
+		break;
+	case V18:
+		DCDC_REG_SET(V18, CUR_LIM_MAX_HV, ((max_ua / 30000) - 1));
+		DCDC_REG_SET(V18, CUR_LIM_MAX_LV, ((max_ua / 30000) - 1));
+		DCDC_REG_SET(V18, CUR_LIM_MIN, ((max_ua / 30000) - 1));
+		break;
+	case V18P:
+		DCDC_REG_SET(V18P, CUR_LIM_MAX_HV, ((max_ua / 30000) - 1));
+		DCDC_REG_SET(V18P, CUR_LIM_MAX_LV, ((max_ua / 30000) - 1));
+		DCDC_REG_SET(V18P, CUR_LIM_MIN, ((max_ua / 30000) - 1));
+		break;
+	default:
+		ret = -ENOSYS;
+		break;
+	}
+
+	return ret;
+}
+
+static int regulator_da1469x_get_current_limit(const struct device *dev,
+					       int32_t *curr_ua)
+{
+	const struct regulator_da1469x_config *config = dev->config;
+	int ret = 0;
+
+	switch (config->rail) {
+	case VDD:
+		*curr_ua = (DCDC_REG_GET(VDD, CUR_LIM_MAX_HV) - 1) * 30000;
+		break;
+	case V14:
+		*curr_ua = (DCDC_REG_GET(V14, CUR_LIM_MAX_HV) - 1) * 30000;
+		break;
+	case V18:
+		*curr_ua = (DCDC_REG_GET(V18, CUR_LIM_MAX_HV) - 1) * 30000;
+		break;
+	case V18P:
+		*curr_ua = (DCDC_REG_GET(V18P, CUR_LIM_MAX_HV) - 1) * 30000;
+		break;
+	default:
+		ret = -ENOSYS;
+	}
+
+	return ret;
+}
+
+static const struct regulator_driver_api regulator_da1469x_api = {
+	.enable = regulator_da1469x_enable,
+	.disable = regulator_da1469x_disable,
+	.count_voltages = regulator_da1469x_count_voltages,
+	.list_voltage = regulator_da1469x_list_voltage,
+	.set_voltage = regulator_da1469x_set_voltage,
+	.get_voltage = regulator_da1469x_get_voltage,
+	.set_current_limit = regulator_da1469x_set_current_limit,
+	.get_current_limit = regulator_da1469x_get_current_limit,
+};
+
+static int regulator_da1469x_init(const struct device *dev)
+{
+	const struct regulator_da1469x_config *config = dev->config;
+
+	regulator_common_data_init(dev);
+
+	if (config->rail == V30) {
+		POWER_REG_SET(LDO_3V0_REF, DT_INST_PROP_OR(DT_NODELABEL(V30),
+							   regulator_v30_ref_bandgap, 0));
+	}
+
+	return regulator_common_init(dev, 0);
+}
+
+#define REGULATOR_DA1469X_DEFINE(inst, id, rail_id)                            \
+	static struct regulator_da1469x_data data_##id;                        \
+                                                                               \
+	static const struct regulator_da1469x_config config_##id = {           \
+		.common = REGULATOR_DT_COMMON_CONFIG_INIT(inst),               \
+		.voltages = voltages_ ## id,                                   \
+		.voltages_count = ARRAY_SIZE(voltages_ ## id),                 \
+		.rail = rail_id,                                               \
+		.flags = (DT_PROP_OR(inst, renesas_regulator_v30_clamp, 0) *   \
+			  DA1469X_MODE_V30_VBAT_CLAMP_ON) |                    \
+			 (DT_PROP_OR(inst, renesas_regulator_v30_vbus, 0) *    \
+			  DA1469X_MODE_LDO_VBUS_ON) |                          \
```

Comment:
For a boolean you don't need OR, so I suggest:

```
DT_PROP(inst, renesas_regulator_v30_vbus) * DA1469X_MODE_LDO_VBUS_ON) 
```

Reply:
done, thanks for a hint